hack/lib/util.sh

```diff
@@ -702,6 +702,23 @@ function kube::util::require-jq {
   fi
 }
 
+# kube::util::read-array
+# Reads in stdin and adds it line by line to the array provided. This can be
+# user instead of "mapfile -t", and is bash 3 compatible.
+#
+# Assumed vars:
+#   $1 (name of array to create/modify)
+#
+# Example usage:
+# kube::util::read-array files < <(ls -1)
+#
+function kube::util::read-array {
+  local i=0
+  unset -v "$1"
+  while IFS= read -r "$1[i++]"; do :; done
+  eval "[[ \${$1[--i]} ]]" || unset "$1[i]"
```

Comment:
what is this guarding against?

Reply:
It's to make sure the last entry in the array isn't empty

Reply:
I feel like this would be more informative to future maintainers as a comment alongside the magic incantation instead of buried in this PR